### PR TITLE
Isolate BIP39 as BitcoinNet.Mnemonic, replace `Pbkdf2` by `Rfc2898DeriveBytes` and remove some unrelated algorithms

### PR DESCRIPTION
**Pull request by osman-turan**
- Isolates BIP39 and related unit tests as `BitcoinNet.Mnemonic` and `BitcoinNet.Mnemonic.Tests` .NET Core 2.1 library projects.
- Replaces `Pbkdf2` class by its BCL counterpart `Rfc2898DeriveBytes` class.
- Removes unrelated algorithms such as `Salsa`, `SCrypt` and `Pbkdf2` implementation.

Resolves: #8, #19

### CODE REVIEW

File: BitcoinNet.Tests/bip39_tests.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 using System.Linq;
```
Comment (vermorel) [inline BitcoinNet.Tests/bip39_tests.cs]: This file should be migrated into its own project `BitcoinNet.Mnemonic.Tests`.

File: BitcoinNet/BouncyCastle/crypto/macs/HMac.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace BitcoinNet.BouncyCastle.Crypto.Macs
     *
     * H(K XOR opad, H(K XOR ipad, text))
     */
-	internal class HMac
+	public class HMac
```
Comment (vermorel) [inline BitcoinNet/BouncyCastle/crypto/macs/HMac.cs]: This class appears to be only used for `Pbkdf2` and BIP39. I suggest to move it to the `BitcoinNet.Mnemonic` project.

File: BitcoinNet/Crypto/Cryptsharp/Pbkdf2.cs [hunk 3]
```diff
@@ -111,7 +111,7 @@ public byte[] Read(int count)
 		/// <param name="iterations">The number of iterations to apply.</param>
 		/// <param name="derivedKeyLength">The desired length of the derived key.</param>
 		/// <returns>The derived key.</returns>
-		internal static byte[] ComputeDerivedKey(IMac hmacAlgorithm, byte[] salt, int iterations,
+		public static byte[] ComputeDerivedKey(IMac hmacAlgorithm, byte[] salt, int iterations,
```
Comment (vermorel) [inline BitcoinNet/Crypto/Cryptsharp/Pbkdf2.cs]: Idem, only used for `BitcoinNet.Mnemonic`, should be move to this project.

**Comment by vermorel** [comment]
Looks good, but there are quite a few things in the `BitcoinNet` project that are only there because they support (indirectly) the BIP39. We should move as much as possible those dependencies to the `BitcoinNet.Mnemonic` project. Thanks!

**Comment by osman-turan** [comment]
`Mnemonic` class suffers same problem as `Script` class due to its namespace. So, in the code I had to use `Mnemonic.Mnemonic` to indicate the class. Do you want me to leave as it is, or have an idea about it?

**Comment by vermorel** [comment]
I suggest to rename the class `Mnemonic` as `MnemonicSequence`.

File: BitcoinNet/BouncyCastle/crypto/macs/HMac.cs (hunk 2 above)
Comment (osman-turan) [reply BitcoinNet/BouncyCastle/crypto/macs/HMac.cs]: I ended up including #19 in this pull-request as well by replacing `Pbkdf2` by `Rfc2898DeriveBytes` (I haven't pushed the update yet). So, only `SCrypt` class uses `Pbkdf2` uses it, now.

Comment (osman-turan) [reply BitcoinNet/BouncyCastle/crypto/macs/HMac.cs]: Never mind. I remembered you want me to remove `SCrypt` along some other unrelated algorithms. I've updated the pull-request.

**Pull request merged**